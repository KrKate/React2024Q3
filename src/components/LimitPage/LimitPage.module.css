.limitContainer {
    display: flex;
    justify-content: center;
    font-size: 18px;
}

label {
    display: flex;
    gap: 0.2rem;
}