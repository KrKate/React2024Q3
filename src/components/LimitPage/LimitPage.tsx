import { useDispatch, useSelector } from 'react-redux';
import styles from './LimitPage.module.css';
import { setLimit } from '../../redux/store/homePageSlice';
import {
  setCurrentPage,
  setTotalPages,
} from '../../redux/store/paginationSlice';
import { AppRootState } from '../../redux/reducers';

function LimitPage() {
  const total = useSelector((state: AppRootState) => state.homePage.total);
  const limit = useSelector((state: AppRootState) => state.homePage.limit);
  const dispatch = useDispatch();
  const handleSelectChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
    dispatch(setLimit(parseInt(e.target.value, 10)));
    dispatch(setCurrentPage(1));
    dispatch(setTotalPages(Math.ceil(total / limit)));
  };

  return (
    <section className={styles.limitContainer}>
      <label htmlFor="limit">
        Items:
        <select
          id="limit"
          onChange={handleSelectChange}
          defaultValue="10"
          data-testid="limit-select"
        >
          <option value="5">5</option>
          <option value="10">10</option>
          <option value="15">15</option>
          <option value="20">20</option>
        </select>
      </label>
    </section>
  );
}

export default LimitPage;
