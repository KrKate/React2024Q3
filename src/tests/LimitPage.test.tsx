import { render } from '@testing-library/react';
import { vi } from 'vitest';
import * as reduxHooks from 'react-redux';
import LimitPage from '../components/LimitPage/LimitPage';

vi.mock('react-redux');
const mockedDispatch = vi.spyOn(reduxHooks, 'useDispatch');
describe('LimitPage component', () => {
  it('should create limit item', () => {
    mockedDispatch.mockResolvedValue(vi.fn());
    const component = render(<LimitPage />);
    expect(component).toMatchSnapshot();
  });
});
