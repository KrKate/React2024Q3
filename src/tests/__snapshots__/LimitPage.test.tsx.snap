// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LimitPage component > should create limit item 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="_limitContainer_f0199d"
      >
        <label
          for="limit"
        >
          Items:
          <select
            data-testid="limit-select"
            id="limit"
          >
            <option
              value="5"
            >
              5
            </option>
            <option
              selected=""
              value="10"
            >
              10
            </option>
            <option
              value="15"
            >
              15
            </option>
            <option
              value="20"
            >
              20
            </option>
          </select>
        </label>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="_limitContainer_f0199d"
    >
      <label
        for="limit"
      >
        Items:
        <select
          data-testid="limit-select"
          id="limit"
        >
          <option
            value="5"
          >
            5
          </option>
          <option
            selected=""
            value="10"
          >
            10
          </option>
          <option
            value="15"
          >
            15
          </option>
          <option
            value="20"
          >
            20
          </option>
        </select>
      </label>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
